Geometry/HGCalMapping/plugins/HGCalMappingTriggerESProducer.cc

```diff
@@ -0,0 +1,144 @@
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+#include "CondFormats/DataRecord/interface/HGCalElectronicsMappingRcd.h"
+
+#include "CondFormats/HGCalObjects/interface/HGCalMappingModuleIndexerTrigger.h"
+#include "CondFormats/HGCalObjects/interface/HGCalMappingCellIndexerTrigger.h"
+#include "CondFormats/HGCalObjects/interface/HGCalMappingParameterHost.h"
+
+#include "DataFormats/HGCalDigi/interface/HGCalElectronicsId.h"
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCScintillatorDetId.h"
+#include "Geometry/HGCalMapping/interface/HGCalMappingTools.h"
+#include <regex>  // regular expression
+
+/**
+   @short plugin parses the module/cell locator files to produce the indexer records
+ */
+class HGCalMappingTriggerESProducer : public edm::ESProducer, public edm::EventSetupRecordIntervalFinder {
+public:
+  explicit HGCalMappingTriggerESProducer(const edm::ParameterSet& iConfig) {
+    //parse the files and hold the list of entities in memory
+    for (const auto& v : {"modules", "si", "sipm"}) {
+      edm::FileInPath fip = iConfig.getParameter<edm::FileInPath>(v);
+      hgcal::mappingtools::HGCalEntityList pmap;
+      pmap.buildFrom(fip.fullPath());
+      parsedMaps_[v] = pmap;
+    }
+
+    setWhatProduced(this, &HGCalMappingTriggerESProducer::produceCellMapIndexer);
+    setWhatProduced(this, &HGCalMappingTriggerESProducer::produceModuleMapIndexer);
+
+    findingRecord<HGCalElectronicsMappingRcd>();
+
+    prepareCellMapperIndexer();
+    prepareModuleMapperIndexer();
+  }
+
+  std::shared_ptr<HGCalMappingModuleIndexerTrigger> produceModuleMapIndexer(const HGCalElectronicsMappingRcd&) {
+    return std::shared_ptr<HGCalMappingModuleIndexerTrigger>(&modIndexer_, edm::do_nothing_deleter());
+  }
+
+  std::shared_ptr<HGCalMappingCellIndexerTrigger> produceCellMapIndexer(const HGCalElectronicsMappingRcd&) {
+    return std::shared_ptr<HGCalMappingCellIndexerTrigger>(&cellIndexer_, edm::do_nothing_deleter());
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::FileInPath>("modules")->setComment("module locator file");
+    desc.add<edm::FileInPath>("si")->setComment("file containing the mapping of the readout cells in Si modules");
+    desc.add<edm::FileInPath>("sipm")->setComment(
+        "file containing the mapping of the readout cells in SiPM-on-tile modules");
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+private:
+  void setIntervalFor(const edm::eventsetup::EventSetupRecordKey&,
+                      const edm::IOVSyncValue&,
+                      edm::ValidityInterval& oValidity) override {
+    oValidity = edm::ValidityInterval(edm::IOVSyncValue::beginOfTime(), edm::IOVSyncValue::endOfTime());
+  }
+
+  void prepareCellMapperIndexer();
+  void prepareModuleMapperIndexer();
+
+  std::map<std::string, hgcal::mappingtools::HGCalEntityList> parsedMaps_;
```

Comment:
OK, no.

Please make three separate data members, instead of going through an `std::map` each time to access one of them.